fri/src/two_adic_pcs.rs

```diff
@@ -24,167 +24,135 @@ use tracing::{info_span, instrument};
 use crate::verifier::{self, FriError};
 use crate::{prover, FriConfig, FriProof};
 
-/// We group all of our type bounds into this trait to reduce duplication across signatures.
-pub trait TwoAdicFriPcsGenericConfig: Default {
-    type Val: TwoAdicField;
-    type Challenge: TwoAdicField + ExtensionField<Self::Val>;
-    type Challenger: FieldChallenger<Self::Val>
-        + GrindingChallenger<Witness = Self::Val>
-        + CanObserve<<Self::FriMmcs as Mmcs<Self::Challenge>>::Commitment>
-        + CanSample<Self::Challenge>;
-    type Dft: TwoAdicSubgroupDft<Self::Val>;
-    type InputMmcs: 'static
-        + for<'a> DirectMmcs<Self::Val, Mat<'a> = RowMajorMatrixView<'a, Self::Val>>;
-    type FriMmcs: DirectMmcs<Self::Challenge>;
+pub struct TwoAdicFriPcs<Val, Dft, InputMmcs, FriMmcs> {
+    // degree bound
+    log_n: usize,
+    dft: Dft,
+    mmcs: InputMmcs,
+    fri: FriConfig<FriMmcs>,
+    _phantom: PhantomData<Val>,
 }
 
-pub struct TwoAdicFriPcsConfig<Val, Challenge, Challenger, Dft, InputMmcs, FriMmcs>(
-    PhantomData<(Val, Challenge, Challenger, Dft, InputMmcs, FriMmcs)>,
-);
-impl<Val, Challenge, Challenger, Dft, InputMmcs, FriMmcs> Default
-    for TwoAdicFriPcsConfig<Val, Challenge, Challenger, Dft, InputMmcs, FriMmcs>
-{
-    fn default() -> Self {
-        Self(PhantomData)
-    }
-}
-
-impl<Val, Challenge, Challenger, Dft, InputMmcs, FriMmcs> TwoAdicFriPcsGenericConfig
-    for TwoAdicFriPcsConfig<Val, Challenge, Challenger, Dft, InputMmcs, FriMmcs>
-where
-    Val: TwoAdicField,
-    Challenge: TwoAdicField + ExtensionField<Val>,
-    Challenger: FieldChallenger<Val>
-        + GrindingChallenger<Witness = Val>
-        + CanObserve<<FriMmcs as Mmcs<Challenge>>::Commitment>
-        + CanSample<Challenge>,
-    Dft: TwoAdicSubgroupDft<Val>,
-    InputMmcs: 'static + for<'a> DirectMmcs<Val, Mat<'a> = RowMajorMatrixView<'a, Val>>,
-    FriMmcs: DirectMmcs<Challenge>,
-{
-    type Val = Val;
-    type Challenge = Challenge;
-    type Challenger = Challenger;
-    type Dft = Dft;
-    type InputMmcs = InputMmcs;
-    type FriMmcs = FriMmcs;
-}
-
-pub struct TwoAdicFriPcs<C: TwoAdicFriPcsGenericConfig> {
-    fri: FriConfig<C::FriMmcs>,
-    dft: C::Dft,
-    mmcs: C::InputMmcs,
-}
-
-impl<C: TwoAdicFriPcsGenericConfig> TwoAdicFriPcs<C> {
-    pub fn new(fri: FriConfig<C::FriMmcs>, dft: C::Dft, mmcs: C::InputMmcs) -> Self {
-        Self { fri, dft, mmcs }
+impl<Val, Dft, InputMmcs, FriMmcs> TwoAdicFriPcs<Val, Dft, InputMmcs, FriMmcs> {
```

Comment:
now that there's only one `Pcs` trait to implement I got rid of `TwoAdicFriPcsConfig`